nengo_gui/static/test/datastore_test.js

```diff
@@ -0,0 +1,201 @@
+// run this file with `mocha datastore_test.js`
+// Will figure out how to integrate with TravisCI later
+
+global.Nengo = {};
+
+require("../datastore")
+var assert = require("assert");
+
+var sim = {
+            time_slider:{
+                first_shown_time:0,
+                shown_time:0,
+                last_time:0,
+                kept_time:2
+            }
+}
+
+// I should initialise the data_store before each test or does it do that already?
+describe("DataStore", function() {
+    var data_store = new Nengo.DataStore(2, sim, 0);
+
+    beforeEach(function() {
+        data_store = new Nengo.DataStore(2, sim, 0);
+    });
+
+    it("accepts data", function() {
+        data_store.push([0.0, 1.1, 1.2]);
+        assert.deepEqual(data_store.data, [[1.1], [1.2]]);
+    });
+
+    it("orders data", function() {
+        data_store.push([1.0, 3.1, 3.2]);
+        data_store.push([0.5, 2.1, 2.2]);
+        assert.deepEqual(data_store.data, [[2.1], [2.2]]);
```

Comment:
Why as `data` only that one row with 2.x? (I suppose, the first value is the time?)